revolv/administrator/views.py

```diff
@@ -29,6 +35,423 @@ def get_context_data(self, **kwargs):
+    """
+    def dispatch(self, request, *args, **kwargs):
+        request = request.GET
+        keys = request.keys()
+        
+        # THESE ARE DUMMY VARIABLES - USE THE REVOLV EARNINGS MODEL INSTEAD
+        payment_service_fee_rate = 0.01
+        revolv_earnings_rate = 0.07
+
+        self.updateFilterVariables(request)
+
+        filtered_dict = self.setUpFilteredDict()      
+        date_info = filtered_dict['date_info']
+        date_list = date_info['date_list']
+
+        cash_in_repayment = filtered_dict['cash_in']['repayments']
+        cash_in_donation = filtered_dict['cash_in']['donations']
+        cash_in_adjustment = filtered_dict['cash_in']['adjustments']
+        cash_out_reinvestment = filtered_dict['cash_out']['reinvestments']
+        cash_out_adjustment = filtered_dict['cash_out']['adjustments']
+
+        temp_start_date = self.start_date
+        temp_end_date = self.moveOneMonthForward(temp_start_date)
+
+        cash_balance = self.calculateCashBalanceFromStart()
+        filtered_project_filter = [proj for proj in self.project_filter if self.hasPayments(proj)]
+        
+        while temp_start_date < self.end_date:
+            net_cash_in = 0
+            net_cash_out = 0
+
+            if (self.end_date < temp_end_date):
+                temp_end_date = self.end_date
+
+            date_list.append(temp_start_date.strftime('%Y-%m-%d'))
+            date = temp_start_date.strftime('%Y-%m-%d')        
+            self.makePaymentTypeKeys(filtered_dict, date)
+
+            cash_in_repayment_date = cash_in_repayment[date]
+            cash_in_donation_date = cash_in_donation[date]
+            cash_in_adjustment_date = cash_in_adjustment[date]
+            cash_out_reinvestment_date = cash_out_reinvestment[date]
+            cash_out_adjustment_date = cash_out_adjustment[date]
+            cash = filtered_dict['cash_balances'][date]
+
+            self.setTimePeriodAdjustments(filtered_dict, date, temp_start_date, temp_end_date)
+            net_cash_in += cash_in_adjustment_date['total']
+            net_cash_out += cash_out_adjustment_date['total']
+
+            for proj in filtered_project_filter:
+                self.makeProjectKeys(filtered_dict, date, title)
+                self.setAllTableValues(filtered_dict, date, proj, temp_start_date, temp_end_date, revolv_earnings_rate, payment_service_fee_rate)
+                
+                net_cash_in += cash_in_repayment_date[proj.title]['retained_ROI']
+                net_cash_in += cash_in_donation_date[proj.title]['retained_donations']
+                net_cash_out += cash_out_reinvestment_date[proj.title]['total']
+            
+            self.setCashBalances(cash, cash_balance, net_cash_in, net_cash_out)
+            cash_balance += net_cash_in - net_cash_out
+
+            temp_start_date = temp_end_date
+            temp_end_date = self.moveOneMonthForward(temp_end_date)
+        
+        return JsonResponse(filtered_dict)
```

Comment:
![screen shot 2015-05-28 at 7 53 18 pm](https://cloud.githubusercontent.com/assets/7706852/7875504/39b6d67a-0573-11e5-8b36-9196ce828b69.png)
I AGREE THIS IS SO MUCH BETTER
this is the light